src/index/base.cpp

```diff
@@ -48,8 +48,6 @@ bool BaseIndex::DB::WriteBestBlock(const CBlockLocator& locator)
 
 BaseIndex::~BaseIndex()
 {
-    Interrupt();
-    Stop();
```

Comment:
You could check here that `m_thread_sync` is not `joinable`, to enforce that subclasses actually stop the thread as they should.

Reply:
This is done by `~thread`, which `std::terminate`s on that occasion.

Reply:
Yes indeed.  Would it make sense to add a brief comment somewhere mentioning this?  It would certainly have helped clarify it for me.